src/index.js

```diff
@@ -67,7 +82,35 @@ exports.sourceNodes = async (
 
   // Merge single and content types and retrieve create nodes
   types.forEach(({ name }, i) => {
+    const { createTypes } = actions;
     const items = entities[i];
+
+    // Search for dynamic zones in all items
+    const dynamicZoneFields = [];
+    const entries = items.flatMap((item) => Object.entries(item));
+    entries.forEach(([field, value]) => {
+      if (isDynamicZone(value)) {
+        // Add to the list of dynamic zone fields if it's not already there
+        if (!dynamicZoneFields.includes(field)) {
+          dynamicZoneFields.push(field);
+        }
+      }
+    });
+
+    // Cast dynamic zone fields to JSON
+    if (dynamicZoneFields.length > 0) {
+      const typeDefs = `
+        type Strapi${capitalize(name)} implements Node {
+          ${dynamicZoneFields.map(
```

Comment:
When doing dynamic creation of types you can use the programmatic api instead to avoid this string concats

here is the ref https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#gatsby-type-builders